Tests/StoreKitUnitTests/StoreMessagesHelperTests.swift

```diff
@@ -33,136 +32,107 @@ class StoreMessagesHelperTests: TestCase {
     override func setUpWithError() throws {
         try super.setUpWithError()
 
+        try AvailabilityChecks.iOS16APIAvailableOrSkipTest()
+
         self.systemInfo = MockSystemInfo(finishTransactions: true)
         self.storeMessagesProvider = MockStoreMessagesProvider()
     }
 
-    @available(iOS 16.4, *)
     func testShowMessagesAfterDeferMessagesAndNotShowingMessagesAutomaticallyShowsAllDeferredMessages() async throws {
-        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
         self.createHelper(showStoreMessagesAutomatically: false)
 
-        let message1 = MockStoreMessage(reason: Message.Reason.billingIssue)
-        let message2 = MockStoreMessage(reason: Message.Reason.priceIncreaseConsent)
+        let message1 = MockStoreMessage(reason: .generic)
+        let message2 = MockStoreMessage(reason: .priceIncreaseConsent)
 
-        try await waitForDeferredMessages(messages: [message1, message2])
+        try await self.waitForDeferredMessages(messages: [message1, message2])
 
         await self.helper.showStoreMessages(types: Set(StoreMessageType.allCases))
 
         expect(message1.displayCalled) == true
         expect(message2.displayCalled) == true
     }
 
-    @available(iOS 16.4, *)
     func testShowMessagesAfterDeferMessagesAndNotShowingMessagesAutomaticallyShowsSpecifiedMessages() async throws {
-        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
         self.createHelper(showStoreMessagesAutomatically: false)
 
-        let message1 = MockStoreMessage(reason: Message.Reason.billingIssue)
-        let message2 = MockStoreMessage(reason: Message.Reason.priceIncreaseConsent)
+        let message1 = MockStoreMessage(reason: .generic)
+        let message2 = MockStoreMessage(reason: .priceIncreaseConsent)
 
-        try await waitForDeferredMessages(messages: [message1, message2])
+        try await self.waitForDeferredMessages(messages: [message1, message2])
 
-        await self.helper.showStoreMessages(types: [StoreMessageType.billingIssue])
+        await self.helper.showStoreMessages(types: [.generic])
 
         expect(message1.displayCalled) == true
         expect(message2.displayCalled) == false
     }
 
-    @available(iOS 16.4, *)
     func testShowMessagesAfterDeferMessagesAndShowingMessagesAutomaticallyDoesNotShowMessages() async throws {
-        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
         self.createHelper(showStoreMessagesAutomatically: true)
 
-        let message1 = MockStoreMessage(reason: Message.Reason.billingIssue)
-        let message2 = MockStoreMessage(reason: Message.Reason.priceIncreaseConsent)
+        let message1 = MockStoreMessage(reason: .generic)
+        let message2 = MockStoreMessage(reason: .priceIncreaseConsent)
```

Comment:
given that we have specific separate logic for `billingIssue`, feels like we should have at least one test that explicitly uses it

Reply:
Yeah we don't have explicit tests for `Message.Reason.messageType` /cc @tonidero

Reply:
Oh ok, will try to add it in a followup PR